src/plugins/discover/public/application/components/sidebar/lib/get_details.ts

```diff
@@ -38,18 +39,30 @@ export function getDetails(
   columns: string[],
   indexPattern?: IndexPattern
 ) {
+  const defaultDetails = {
+    error: '',
+    exists: 0,
+    total: 0,
+    buckets: [],
+    columns,
```

Comment:
Should we include columns as part of the defaultDetails?

Reply:
The reason this reads weird is that `columns` is just a value we pass and return without using... which is a bit weird. I can refactor to remove it from this function's input and output and just pass it as a standalone property to `DiscoverFieldDetails` which would be the more sensible thing to do.